train.py

```diff
@@ -51,6 +51,7 @@ def train(hyp, opt, device, tb_writer=None):
     init_seeds(2 + rank)
     with open(opt.data) as f:
         data_dict = yaml.load(f, Loader=yaml.FullLoader)  # model dict
+    check_dataset(data_dict)  # check
```

Comment:
@NanoCode012 do you know how I should handle this for DDP? This function check_dataset() autodownloads a dataset if not found locally, just like attempt_download() for the models, but of course we only want it to run once even in DDP.

Reply:
You can add a “if rank in [-1,0]” condition check for running on only one GPU once. 

However in this case, torch_distributed_zero_first looks like the better one to use(as you’ve done), as it blocks the other GPU from continuing, as we want to check dataset before proceeding.

Reply:
Ah thanks! Ok it seems everything is all set then...